tests/qos/test_buffer.py

```diff
@@ -0,0 +1,848 @@
+
+testparam_headroom_override = None
+testparam_lossless_pg = None
+
+buffer_model_dynamic = True
+
+def detect_buffer_model(duthost):
+    global buffer_model_dynamic
+    buffer_model = duthost.shell('redis-cli -n 4 hget "DEVICE_METADATA|localhost" buffer_model')['stdout']
+    buffer_model_dynamic = buffer_model == 'dynamic'
+
+
+def detect_ingress_pool_number(duthost):
+    global default_ingress_pool_number
+    pools = duthost.shell('redis-cli -n 4 keys "BUFFER_POOL|ingress*"')['stdout']
+    default_ingress_pool_number = len(pools.split())
+
+
+def load_lossless_headroom_data(duthost):
+    global default_lossless_headroom_data
+    if not default_lossless_headroom_data:
+        dut_hwsku = duthost.facts["hwsku"]
+        dut_platform = duthost.facts["platform"]
+        skudir = "/usr/share/sonic/device/{}/{}/".format(dut_platform, dut_hwsku)
+        lines = duthost.shell('cat {}/pg_profile_lookup.ini'.format(skudir))["stdout"]
+        default_lossless_headroom_data = {}
+        for line in lines.split('\n'):
+            if line[0] == '#':
+                continue
+            tokens = line.split()
+            speed = tokens[0]
+            cable_length = tokens[1]
+            size = tokens[2]
+            xon = tokens[3]
+            xoff = tokens[4]
+            if not default_lossless_headroom_data.get(speed):
+                default_lossless_headroom_data[speed] = {}
+            default_lossless_headroom_data[speed][cable_length] = {'size': size, 'xon': xon, 'xoff': xoff}
+        default_lossless_headroom_data = default_lossless_headroom_data
+
+
+def load_test_parameters(duthost):
+    global default_cable_length_list
+    global testparam_headroom_override
+    global testparam_lossless_pg
+    with open(r"qos/files/dynamic_buffer_param.json") as file:
+        params = json.load(file)
+        logging.info("Got test parameters {}".format(params))
+        asic_type = duthost.facts['asic_type']
+        vendor_specific_param = params[asic_type]
+        default_cable_length_list = vendor_specific_param['default_cable_length']
+        testparam_headroom_override = vendor_specific_param['headroom-override']
+        testparam_lossless_pg = vendor_specific_param['lossless_pg']
+
+
+@pytest.fixture(scope="module", autouse=True)
+def setup_module(duthost):
+    detect_buffer_model(duthost)
+    if buffer_model_dynamic:
+        detect_ingress_pool_number(duthost)
+        load_lossless_headroom_data(duthost)
+        load_test_parameters(duthost)
+
+        logging.info("Got cable length: default {}".format(default_cable_length_list))
+        logging.info("Got ingress pool number {}".format(default_ingress_pool_number))
+        logging.info("Got lossless headroom data {}".format(default_lossless_headroom_data))
+    else:
+        pytest.skip("Dynamic buffer isn't enabled, skip the test")
+
+    yield
+
+
+def check_pool_size(duthost, ingress_lossless_pool_oid, **kwargs):
+    if duthost.facts['asic_type'] == 'mellanox':
```

Comment:
need more information on this check. what is the old comparing to new as both are used

Reply:
I don't quite understand this comment. Could you please elaborate?